### ISSUE TITLE
Cmake config is broken.

### ISSUE THREAD

**Issue by FrankXie05** [hyperrealm/libconfig#231]
Hi, I'm one of the maintainers of vcpkg, and we recently received a question about libconfg using cmake.
Issue:
There seems to be this problem with the exported cmake usage after we installed libconfig, it fails to link against `libconfig++`.
Run command `./vcpkg install libconfig:x64-windows --head` and output:
```
libconfig provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(libconfig CONFIG REQUIRED)
    target_link_libraries(main PRIVATE libconfig::libconfig libconfig::libconfig++)
```

For this reason, I made a patch to separate `libconfig` and `libconfig++`. 

However, from the code structure, `libconfig` does not seem to be separated, but is linked to `libconfig++`.
BUt, it was broken.
https://github.com/hyperrealm/libconfig/blob/4f13b7f5152427b5e511a09617ca2d9ff63f7800/lib/CMakeLists.txt#L144
https://github.com/hyperrealm/libconfig/blob/4f13b7f5152427b5e511a09617ca2d9ff63f7800/lib/CMakeLists.txt#L152

Reproduce steps:
`CMakeLists.txt`
```
cmake_minimum_required(VERSION 3.0)
project(test)
add_executable(main main.cpp)
find_package(libconfig CONFIG REQUIRED)
target_link_libraries(main PRIVATE libconfig::libconfig libconfig::libconfig++)
```
Error:
```
CMake Error at CMakeLists.txt:5 (target_link_libraries):
  Target "main" links to:

    libconfig::libconfig++

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
```
Related vcpkg issue and PR:
https://github.com/microsoft/vcpkg/issues/30906
https://github.com/microsoft/vcpkg/pull/30980

@hyperrealm

**Comment by dg0yt** [comment]
This question drags in a lot of vcpkg particularities. More general (https://github.com/microsoft/vcpkg/pull/30980#discussion_r1172147992), please solve this problem and inconsistency:

Target `libconfig++` is in namespace and directory `libconfig` but in package `libconfig++`.
This makes the target unreachable.
- `find_package(libconfig++)` doesn't work.
- `find_package(libconfig)` doesn't provide the target.

Consistent patterns would be:
- Target `libconfig++` is in namespace `libconfig` and in package `libconfig`.
- Target `libconfig++` is in namespace `libconfig++` and in package `libconfig++`.

I would recommend the first alternative, in particular because they are connected.

**Comment by mllusar** [comment]
is there any way to get around the problem until this issue is closed?

**Comment by mllusar** [comment]
Hey ! Is there any new about this issue ? I really need to use libonfig++ under Cmake with vcpkg.

I'm not an advanced programmer, but it seams that all we need need is to rename the namespace.

**Comment by FrankXie05** [comment]
@mllusar You can use my PR temporarily https://github.com/microsoft/vcpkg/pull/30980

**Comment by dg0yt** [comment]
> * Target `libconfig++` is in namespace, config and directory `libconfig`.

Implemented in #234.
